nuclear/message/include/message/conversion/proto_matrix.hpp

```diff
@@ -1427,8 +1427,9 @@ namespace message::conversion {
         proto.mutable_v()->resize(vector.size());
 
         // Copy the data across
-        Eigen::Map<::message::conversion::math::cvec>(reinterpret_cast<uint8_t*>(proto.mutable_v()->data()),
-                                                      proto.v().size()) = vector;
+        Eigen::Map<::message::conversion::math::cvec>(
+            const_cast<uint8_t*>(reinterpret_cast<const uint8_t*>(proto.mutable_v()->data())),
```

Comment:
This deserves scrutiny. @TrentHouliston is this undefined??

Reply:
This feels dirty, pretty sure it's just the Eigen map type that should be const

Reply:
Actually that won't work either, Why is protobuf returning a const for it's mutable_v data now? when did that change?

Reply:
I am looking everywhere for a definition of `mutable_v`, but I can't find anything. This class confuses me hahaha

Reply:
@KipHamiltons `mutable_v` is defined [here](https://developers.google.com/protocol-buffers/docs/reference/cpp-generated#embeddedmessage) with foo being [v](https://github.com/NUbots/NUbots/blob/main/nuclear/message/proto/Vector.proto#L20).

Reply:
Thanks for the links! They pointed me in the right direction. Given that it's only not compiling on clang, it seems to me that protobuf didn't change their interface, but the interaction with clang12 has some unknown idiosyncrasy. I'm going to try updating our protobuf version.

Reply:
It builds without the const_cast on the the newest protobuf release. I'll submit a PR to update that, then revert these changes.